### PR TITLE
cache dependencies [actions]

### PR DESCRIPTION
**Pull request by arunavabasu-03**
Cacheing setup actions

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    steps:
+      - name: Check out Git repository
+        uses: actions/checkout@v2
+      - name: Set up Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: 14
+      - name: Cache the dependencies
+        uses: c-hive/gha-yarn-cache@v2
```
Comment (suzil) [inline .github/workflows/ci.yml:25]: We don't need another action to setup dependency caching because the setup-node action will do this for us if we specify `cache: yarn`: https://github.com/actions/setup-node#caching-global-packages-data

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+jobs:
+  test:
+    name: 🧪 Test
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out Git repository
+        uses: actions/checkout@v2
+      - name: Set up Node.js
+        uses: actions/setup-node@v1
```
Comment (suzil) [inline .github/workflows/ci.yml:21]: Might as well go ahead and upgrade this to v3 in case the caching doesn't work for this version.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+          cache: 'yarn'
+      - name: Install Node.js dependencies
+        run: yarn install --frozen-lockfile
+      - name: Build 
+        run: yarn build
+      - name: Artifact upload
+        uses: actions/upload-artifact@v3
+        with:
+          name: Radis App
+          path: ./frontend/build
```
Comment (suzil) [inline .github/workflows/ci.yml:54]: I don't think we need both this and https://github.com/suzil/radis-app/blob/main/.github/workflows/main.yml#L12.

Also, there's no reason to upload an artifact to GitHub as we are uploading the artifact to S3 (indirectly through the AWS CDK).

Comment (arunavabasu-03) [reply .github/workflows/ci.yml:54]: ok @suzil

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -11,22 +10,23 @@ on:
...
     steps:
       - name: Check out Git repository
         uses: actions/checkout@v2
       - name: Set up Node.js
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v3
         with:
-          node-version: 14
-
+          node-version: 16
```
Comment (suzil) [inline .github/workflows/ci.yml:23]: If you upgrade node to 16 here, would you also update it in `main.yml` so all of our builds use the same node version?

File: frontend/src/App.test.tsx [hunk 5]
```diff
@@ -1,9 +1,8 @@
-import { render, screen } from "@testing-library/react";
+/* eslint-disable @typescript-eslint/no-empty-function */
+/* eslint-disable react/react-in-jsx-scope */
+import { render } from "@testing-library/react";
 import App from "./App";
-
-test("renders learn react link", () => {
-  // eslint-disable-next-line react/react-in-jsx-scope
+//@ts-ignore
```
Comment (suzil) [inline frontend/src/App.test.tsx:5]: Why do we need the ts-ignore's? We were running type checks before so I'm confused as to why something would stop working when updating GitHub Actions workflows.

File: frontend/src/components/CalcSpectrum.tsx [hunk 6]
```diff
@@ -160,7 +160,9 @@ export const CalcSpectrum: React.FC = () => {
   };
   const databaseWatch = watch("database");
   const modeWatch = watch("mode");
-
+  // @ts-ignore
```
Comment (suzil) [inline frontend/src/components/CalcSpectrum.tsx:163]: why is the ts-ignore needed?

File: frontend/src/setupTests.ts [hunk 7]
```diff
@@ -3,3 +3,7 @@
 // expect(element).toHaveTextContent(/react/i)
 // learn more: https://github.com/testing-library/jest-dom
 import "@testing-library/jest-dom";
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+//@ts-ignore
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+window.URL.createObjectURL = function () {};
```
Comment (suzil) [inline frontend/src/setupTests.ts:9]: Why is this needed?

Comment (arunavabasu-03) [reply frontend/src/setupTests.ts:9]: so basically when we run the test we have to call this  `createObjectURL`. otherwise test will fail.

Comment (suzil) [reply frontend/src/setupTests.ts:9]: Screen Shot 2022-08-01 at 3 44 28 PM
gotcha, I was able to remove the ts-ignore by doing this

Comment (suzil) [reply frontend/src/setupTests.ts:9]: also the above function doesn't need the `no-empty-function` eslint ignore so that can also be removed

File: frontend/src/setupTests.ts [hunk 8]
```diff
@@ -3,3 +3,6 @@
 // expect(element).toHaveTextContent(/react/i)
 // learn more: https://github.com/testing-library/jest-dom
 import "@testing-library/jest-dom";
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+//@ts-ignore
```
Comment (suzil) [inline frontend/src/setupTests.ts:7]:
```suggestion
```
You should be able to remove this now.

**Comment by arunavabasu-03** [comment]
@suzil , i think all things are ok

**Review by suzil — APPROVED** [review]
nice!
